lib/blocks.php

```diff
@@ -528,3 +528,66 @@ function _gutenberg_footnotes_force_filtered_html_on_import_filter( $arg ) {
 add_action( 'init', '_gutenberg_footnotes_kses_init' );
 add_action( 'set_current_user', '_gutenberg_footnotes_kses_init' );
 add_filter( 'force_filtered_html_on_import', '_gutenberg_footnotes_force_filtered_html_on_import_filter', 999 );
+
+/**
+ * Filters the arguments for registering a post type adding the footnotes support by default.
+ *
+ * @access private
+ *
+ * @param array  $args      Array of arguments for registering a post type.
+ *                          See the register_post_type() function for accepted arguments.
+ */
+function _gutenberg_add_footnotes_post_type_support( $args ) {
+	if ( ! empty( $args['supports'] ) ) {
+		$supports = &$args['supports'];
+		if ( in_array( 'editor', $supports, true ) && in_array( 'custom-fields', $supports, true ) && in_array( 'revisions', $supports, true ) && ! in_array( 'footnotes', $supports, true ) ) {
+			$supports[] = 'footnotes';
+		}
+	}
+
+	return $args;
+}
+
+/**
+ * Adds the footnotes support to the post type by default.
+ */
+add_filter(
+	'register_post_type_args',
+	'_gutenberg_add_footnotes_post_type_support'
+);
+
+/**
+ * Registers the footnotes meta field for post types that support it.
+ *
+ * @internal
+ *
+ */
+function _gutenberg_register_footnotes_meta_field() {
+	$post_types = get_post_types(
+		array(
+			'show_in_rest' => true,
+			'public'       => true,
+		)
+	);
```

Comment:
For CPT's `publicly_queryable` is used to determine whether a post is available on the front end of the site. It's possible for a post type to be `public` without being `publicly_queryable`. Yup, really.

This is the safest way of determining if a post type is really public.

```suggestion
	$post_types = get_post_types( array( 'show_in_rest' => true ) );
	$post_types = array_filter( $post_types, 'is_post_type_viewable' );
```